doc/cookbook/source/examples/gaussian_processes/gaussian_process_regression.rst

```diff
@@ -0,0 +1,59 @@
+
+.. math::
+	f(\mathbf{x})\sim\mathcal{GP}(m(\mathbf{x}), k(\mathbf{x},\mathbf{x}))
+
+Any finite set of such function values :math:`\mathbf{f}` is jointly Gaussian with mean specified by the mean function evaluated at the inputs :math:`\mathbf{x}_i` and covariance matrix :math:`\mathbf{C}` with entries :math:`\mathbf{C}_{ij}=k(\mathbf{x}_i,\mathbf{x}_j)`.
+
+In our regression model, we take our target variable :math:`\mathbf{y}` to be our (latent) function values :math:`\mathbf{f}` plus Gaussian noise:
+
+.. math::
+	p(\mathbf{y}|\mathbf{f})=\mathcal{N}(\mathbf{f},\sigma^2\mathbf{I})
+
+See :cite:`Rasmussen2005GPM` for a comprehensive treatment of Gaussian Processes.
+
+-------
+Example
+-------
+
+Imagine we have files with training and test data. We create `CDenseFeatures` (here 64 bit floats aka RealFeatures) and :sgclass:`CRegressionLabels` as:
+
+.. sgexample:: gaussian_process_regression.sg:create_features
+
+To fit the input (training) data :math:`\mathbf{X}`, we have to choose an appropriate :sgclass:`CMeanFunction` and  :sgclass:`CKernel` and instantiate it. Here we use a basic :sgclass:`CZeroMean` and a :sgclass:`CGaussianKernel` with width 1.
+
+.. sgexample:: gaussian_process_regression.sg:create_appropriate_kernel_and_mean_function
+
+We need to specify the inference method to find the posterior distribution of the function values :math:`\mathbf{f}`. Here we choose to perform exact inference with an instance of :sgclass:`CExactInferenceMethod` and pass it the chosen kernel, the training features, the mean function, the labels and an instance of :sgclass:`CGaussianLikelihood`, to specify the distribution of the targets/labels as above. Finally we generate a CGaussianProcessRegression class to be trained.
+
+.. sgexample:: gaussian_process_regression.sg:create_instance
+
+In order to give a prediction :math:`y^*` on an unseen input :math:`\mathbf{x}^*` we have to integrate over the function values:
+
+.. math::
+	p(y^*|\mathbf{x}^*, \mathbf{y}, \mathbf{X})=\int p(\mathbf{y}^*|f^*)p(f^*|\mathbf{x}^*, \mathbf{f})p(\mathbf{f}|\mathbf{y}, \mathbf{X})d\mathbf{f}df^*
+
+Then we can apply the resulting distribution to test data to get the predicted :sgclass:`CRegressionLabels`.
+
+.. sgexample:: gaussian_process_regression.sg:train_and_apply
+
+The prediction above is based on arbitrary hyperparameters :math:`\boldsymbol{\theta}`: kernel width :math:`\tau=1`, kernel scaling :math:`\gamma=1` and observation noise :math:`\sigma^2=1`. However, we can optimize the marginal likelihood (or model evidence) :math:`p(\mathbf{y}|\mathbf{X}, \boldsymbol{\theta})` w.r.t. :math:`\boldsymbol{\theta}` to get the optimal hyperparameters.
```

Comment:
"However, we can optimize" -> "We can also learn these parameters via optimizing the marginal likelihood"